dali/kernels/common/utils.h

```diff
@@ -32,12 +33,27 @@ T Permute(const T& in, const U& permutation) {
 
 template <typename Shape>
 DALI_HOST_DEV
+void CalcStrides(Shape &strides, const Shape& shape) {
+  strides[strides.size() - 1] = 1;
+  for (int d = strides.size() - 2; d >= 0; d--) {
+    strides[d] = strides[d + 1] * shape[d + 1];
+  }
+}
+
+template <typename OutStrides, typename Shape>
+DALI_HOST_DEV
+void CalcStrides(OutStrides &out_strides, const Shape& shape) {
+  Shape strides = shape;
+  CalcStrides(strides, shape);
```

Comment:
I would add a comment that you cannot call just:
```
CalcStrides(shape, shape);
```
As it would mess up the input.

Reply:
but shape is read-only. I'd add a comment saying that strides has the same size as shape